File: leetcode/easy/278. First_Bad_Version/sol.py
from typing import int

def firstBadVersion(self, n: int) -> int:
    start = 1
    end = n
    
    while start <= end:
        mid = start + int((end-start)/2)
        if isBadVersion(mid):
            end = mid - 1
        else:
            start = mid + 1
    
    return start

File: leetcode/easy/278. First-Bad-Version/sol.js
var solution = function (isBadVersion) {
  /**
   * @param {integer} n Total versions
   * @return {integer} The first bad version
   */
  return function (n) {
    let start = 1,
      end = n;
    let middle = 0;
    while (start <= end) {
      middle = parseInt((start + end) / 2);
      if (isBadVersion(middle)) {
        end = middle - 1;
      } else {
        start = middle + 1;
      }
    }
    return start;
  };
};
